modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/EqualsAndHashCodeModelValidator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.validation;
+
+import static org.mule.runtime.extension.api.util.ExtensionMetadataTypeUtils.getType;
+import static java.lang.String.format;
+import static java.lang.reflect.Modifier.isAbstract;
+import static org.mule.runtime.module.extension.internal.loader.validation.ModelValidationUtils.isASTMode;
+import static org.mule.runtime.module.extension.internal.loader.validation.ModelValidationUtils.isCompiletime;
+
+import org.mule.metadata.api.model.ObjectType;
+import org.mule.metadata.api.visitor.MetadataTypeVisitor;
+import org.mule.runtime.api.meta.NamedObject;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.meta.model.config.ConfigurationModel;
+import org.mule.runtime.api.meta.model.connection.ConnectionProviderModel;
+import org.mule.runtime.api.meta.model.connection.HasConnectionProviderModels;
+import org.mule.runtime.api.meta.model.util.ExtensionWalker;
+import org.mule.runtime.extension.api.loader.ExtensionModelValidator;
+import org.mule.runtime.extension.api.loader.Problem;
+import org.mule.runtime.extension.api.loader.ProblemsReporter;
+
+/**
+ * Validates that POJOs parameter overrides Equals and HashCode
+ *
+ * @since 4.2
+ */
+public class EqualsAndHashCodeModelValidator implements ExtensionModelValidator {
+
+  @Override
+  public void validate(ExtensionModel extensionModel, ProblemsReporter problemsReporter) {
+    if (!isCompiletime(extensionModel) || isASTMode(extensionModel)) {
+      return;
+    }
+
+    new ExtensionWalker() {
+
+      @Override
+      protected void onConfiguration(ConfigurationModel model) {
+        model.getAllParameterModels().forEach(parameterModel -> {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove useless curly braces around statement ([squid:S1602](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1602))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7618&issues=AWh3zunwotIXnmsUD52M&open=AWh3zunwotIXnmsUD52M)

[ISSUE_KEY]: AWh3zunwotIXnmsUD52M